docs/sql-data-sources-jdbc.md

```diff
@@ -156,6 +156,20 @@ the following case-insensitive options:
      </td>
   </tr>
 
+  <tr>
+     <td><code>preActions</code></td>
+     <td>
+       Custom queries which you want to run before reading data from JDBC or writing data to JDBC. Only DDL or DML (insert/update/delete) are allowed. It is called per DataFrame, not per session. You can specify multiple queries separated by semicolon. When exceptions occur in preActions, the queries in preActions will be rollbacked. 
```

Comment:
>> It is called per DataFrame, not per session.

I think this statement looks ambiguous. What does `per DataFrame` mean? Probably, `per session` means `sessionInitStatement` though.

One more question; is there any usecase `preActions` can handle and `sessionInitStatement` cannot?

Reply:
> ~One more question; is there any usecase `preActions` can handle and `sessionInitStatement` cannot?~

Ah, I see. `sessionInitStatement` is executed on executor-sides and It seems `preActions` is proposed to be executed in a driver side. Probably, you'd be better to describe what's a difference between them in the doc clearly.

Reply:
You are right.. this logic should not be implemented here.

I moved it to `createRelation` to make sure that it is called only once and to have consistency in the timing for both read path and write path.
I also updated the document to describe the differences.